custom_python_actions/utils.py

```diff
@@ -0,0 +1,23 @@
+import time
+
+import github3
+
+
+def download_gh_file(repo: github3.github.repo, file_path: str) -> str:
+    # sometimes the request does not work the first time, so set a retry
+    for attempt in range(5):
+        try:
+            file_content = repo.file_contents(file_path)
+            break
+        except ConnectionResetError as error:
+            if attempt < 4:
+                time.sleep(3)
```

Comment:
Should we have exponential back-off here? Instead of waiting three seconds every time?

Reply:
Yes, I looked into that a little... the python requests retry method only works with specific response error codes, not with a specific type of exception. So this seemed like the simplest way to do retry it. I could implement my own exponential back-off handler but I doubt this will happen very often.

Reply:
Also if you could log that the we are waiting to download artifacts, that would be great as we would know it is not hanging but actually waiting for a bit longer!